lib/dataset/bin.cpp

```diff
@@ -413,13 +414,15 @@ class TargetBinBuilder {
   void group(const Variable &groups) {
     const auto dim = groups.dims().inner();
     m_dims.addInner(dim, groups.dims()[dim]);
-    m_actions.emplace_back(AxisAction::Group, dim, groups);
+    const auto con_groups = scipp::variable::as_contiguous(groups, dim);
+    m_actions.emplace_back(AxisAction::Group, dim, con_groups);
   }
 
   void bin(const Variable &edges) {
     const auto dim = edges.dims().inner();
     m_dims.addInner(dim, edges.dims()[dim] - 1);
-    m_actions.emplace_back(AxisAction::Bin, dim, edges);
+    const auto con_edges = scipp::variable::as_contiguous(edges, dim);
+    m_actions.emplace_back(AxisAction::Bin, dim, con_edges);
```

Comment:
These two functions, `group` and `bin` were the only places I used extra `as_contiguous` to fix the issue.

Reply:
This is technically not equivalent to the current contract: We usually guarantee that the output holds a reference to the input variables representing edges or groups. With this change we can get a copy instead.